source/solvers/multiphysics_interface.cc

```diff
@@ -113,5 +173,62 @@ MultiphysicsInterface<dim>::get_filtered_phase_fraction_gradient_dof_handler()
     ->get_filtered_phase_fraction_gradient_dof_handler();
 }
 
+template <int dim>
+void
+MultiphysicsInterface<dim>::inspect_multiphysics_models_dependencies(
+  const SimulationParameters<dim> &nsparam)
+{
+  bool buoyancy_force_enabled = nsparam.multiphysics.buoyancy_force;
+  bool heat_transfer_enabled  = nsparam.multiphysics.heat_transfer;
+  bool marangoni_effect_enabled =
+    nsparam.multiphysics.vof_parameters.surface_tension_force
+      .enable_marangoni_effect;
+  bool surface_tension_force_enabled =
+    nsparam.multiphysics.vof_parameters.surface_tension_force.enable;
+  bool fluid_dynamics_enabled = nsparam.multiphysics.fluid_dynamics;
+  bool interface_sharpening_enabled =
+    nsparam.multiphysics.vof_parameters.sharpening.enable;
+  bool VOF_enabled = nsparam.multiphysics.VOF;
+
+  // To avoid getting unused parameter warning
+  _unused(buoyancy_force_enabled && heat_transfer_enabled &&
+          marangoni_effect_enabled && surface_tension_force_enabled &&
+          fluid_dynamics_enabled && interface_sharpening_enabled &&
+          VOF_enabled);
+
+  // Dependence of buoyant force on fluid dynamics
+  Assert(!(buoyancy_force_enabled == true && fluid_dynamics_enabled == false),
+         BuoyancyWithoutFluidDynamicsError(buoyancy_force_enabled));
```

Comment:
This is exactly how these should be used. Very, very good work. This is very robust programming, good job.